byway/src/main/webapp/routepage.js

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+if (document.readyState === 'loading') {  // Loading hasn't finished yet
+  document.addEventListener('DOMContentLoaded', getRecs());
```

Comment:
Notice that what you're doing here is equivalent to this:
```js
let recs = getRecs();
document.addEventListener('DOMContentLoaded', recs);
```
The function you provide is called eagerly when you provide parenthesis like this. If you want to actually register a function as an event handler, you need to leave off the parens in order to refer to the function itself rather than calling it.

Reply:
Ah, I see. Thanks for pointing that out.